R/secret_santa.R

```diff
@@ -5,19 +5,30 @@
 #' @param names Vector of names.
 #' @param seed Random seed. Default 2512.
 #' @return Data frame of names
+#' @examples
+#' secret_santa(c("Dasher", "Dancer", "Prancer", "Vixen"))
 #' @export
 
 secret_santa <- function(names, seed = 2512) {
-  if (length(unique(names)) != length(names)) {
+  if (anyDuplicated(names)) {
     stop("Non-unique names provided - do you need to add a surname?")
   }
-  set.seed(seed)
-  random_names <- sample(names)
-  output <- data.frame(
-    gift_giver = random_names,
-    gift_recipient = random_names[c(2:length(random_names), 1)]
+
+  if (length(names) <= 3) {
```

Comment:
Feel like it should it be an error if 1 or 2 people, but a warning if 3? Might still want to use it for gift assignment even if it's not a secret.

Reply:
Sure thing https://github.com/nrennie/christmas/pull/1/commits/ae317f8baeb839b8492d3b1d764851a029e96a93